tests/test_faster_rcnn_util.py

```diff
@@ -272,3 +293,199 @@ def test_convert2canonical_batch():
         ],
     ]
     assert same_list_of_list_of_bboxes(actual_result, expected_result)
+
+
+@patch("datasetinsights.estimators.faster_rcnn.get_world_size")
+def test_reduce_dict_non_dist(test_patch):
+    """test reduce dict non dist."""
+    input_dict = {
+        "loss_classifier": np.nan,
+        "loss_box_reg": np.nan,
+        "loss_objectness": np.nan,
+        "loss_rpn_box_reg": np.nan,
+    }
+    test_patch.return_value = 1
+    expected_result = reduce_dict(input_dict)
+    assert input_dict == expected_result
+
+
+@patch("datasetinsights.estimators.faster_rcnn.get_world_size")
+@patch("datasetinsights.estimators.faster_rcnn.dist.all_reduce")
+def test_reduce_dict_dist(mock_all_reduce, mock_get_world_size):
+    """test reduce dict dist."""
+    input_dict = {
+        "loss_classifier": torch.Tensor([0.4271]),
+        "loss_box_reg": torch.Tensor([0.4271]),
+        "loss_objectness": torch.Tensor([0.4271]),
+        "loss_rpn_box_reg": torch.Tensor([0.4271]),
+    }
+    mock_get_world_size.return_value = 3
+    mock_all_reduce.return_value = MagicMock()
+
+    actual_result = reduce_dict(input_dict)
+    expected_result = {
+        "loss_classifier": 0.1423666626214981,
+        "loss_box_reg": 0.1423666626214981,
+        "loss_objectness": 0.1423666626214981,
+        "loss_rpn_box_reg": 0.1423666626214981,
+    }
+
+    assert all(
+        [
+            actual_result["loss_classifier"].item()
+            == expected_result["loss_classifier"]
+            and actual_result["loss_box_reg"].item()
+            == expected_result["loss_box_reg"]
+            and actual_result["loss_classifier"].item()
+            == expected_result["loss_classifier"]
+            and actual_result["loss_objectness"].item()
+            == expected_result["loss_objectness"]
+            and actual_result["loss_rpn_box_reg"].item()
+            == expected_result["loss_rpn_box_reg"]
+        ]
+    )
+
+
+def test_canonical2list():
+    """test canonical to list."""
+    bbox = BBox2D(label=0, x=10, y=10, w=10, h=10)
+    actual_result = canonical2list(bbox)
+    expected_result = [0, 1.0, 10, 10, 10, 10]
+    assert actual_result == expected_result
+
+
+def test_list2canonical():
+    """test list to canonical."""
+    input_list = [0, 1.0, 10, 10, 10, 10]
+    actual_result = list2canonical(input_list)
+
+    assert all(
```

Comment:
[nit] why not use [asserlistequal](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertListEqual) or simply` assert input_list == actual_result`?